### PR TITLE
Switch to new versioning method; close #58.

### LINKED ISSUE #58: Possible Versioning Change
(Continuing on from discussion on #57).

How about 'Ver. x.x(.x), for SDL 2.0.1 - 2.0.5'?  As there's at least some effort for backwards compatibility going on, having the latest SDL version in the main version number seems counter-intuitive.

Something I would additionally like to see would be a unified set of guidelines for what increments which version digit, perhaps even adopting the [SemVer](semver.org) standard.

At some point, to facilitate added functionality, I may want to try to move the `obj:getWhatever()` interface to a `obj.whatever` implicit-getter style interface (probably using [lstruct](https://gitlab.com/axtel-sturnclaw/lstruct), once I get the C API production-ready).  A big 'Breaking Changes' number to increment would be a virtual necessity with those kinds of changes.

### PR DESCRIPTION
**Pull request by Web-eWorks**
Regarding #58, I have implemented a move to SemVer and an independent version number for the binding.

Still undecided if the patch field is necessary - most changes happening are enough to skip straight to incrementing the minor version.

I moved away from the 'long constant' style of querying the SDL version.  Now the SDL version is queried via `SDL.version.x`, where `x` is either `major`, `minor`, or `patch`.  The binding version is in the `SDL.binding` table.

Thoughts, questions?  Did I miss anything?

Also, what's the `rocks/config.h` file for, and do I have to manually update it?

### CODE REVIEW

**Comment by Tangent128** [comment]
Ah shoot, `rocks/config.h` didn't get updated in the release, did it.

That file is used when building via Luarocks, so that cmake is not needed as a dependency.

That will need to be updated appropriately before this is merged, yes, especially if introducing new #defines.

------

I like the direction of this patchset, but I have notes:

* We should keep `VERSION_MAJOR` and friends around, as removing them would be an API breakage. We can change their meaning in backwards-compatible fashion, though.
* As far as the Luarocks repository is concerned, our latest version is `2.0.5.6.0`; the next release's version should be greater than that rather than reset to `1.0`. Either `2.0.6` or (if we go ahead and ditch the patch level) `2.1`.

**Comment by Web-eWorks** [comment]
Alright, will get a commit pushed later to update that.  Will also amend the versioning to `MAJOR.MINOR`.

You're completely right about the backwards compatibility.  I would suggest marking the SDL version fields as deprecated in the docs, and just dropping the binding ones, as the meaning and names of the `VERSION_BINDING`s are going to change anyway.

I'm hesitant to go with `2.0.6` for the version number, as Lua-SDL2 releases are probably not going to correspond that closely with SDL2 releases, but `2.1` seems perfect.

Also, on a bit of a tangent, would you want to update the description on the Luarocks page?  It seems a little sparse. Maybe merge the the 'Features' section in the updated README into the page?

File: src/SDL.c [hunk 1]
```diff
@@ -485,9 +485,20 @@ luaopen_SDL(lua_State *L)
 	tableSetInt(L, -1, "VERSION_MAJOR", ver.major);
 	tableSetInt(L, -1, "VERSION_MINOR", ver.minor);
 	tableSetInt(L, -1, "VERSION_PATCH", ver.patch);
+	lua_newtable(L);
 
-	tableSetInt(L, -1, "VERSION_BINDING", VERSION_BINDING);
```
Comment (Tangent128) [inline src/SDL.c]: Removing these fields would be a breaking change, even if they probably don't see use in practice;

I suggest hardcoding them to "4" and "1" respectively, so old code can recognize it as a "newer but compatible version", while new code can use the "binding" field to get the real version in the new numbering system.

**Comment by Tangent128** [comment]
Finally had time to look over this and your other branch, had two small kickbacks before I merge.

**Comment by Web-eWorks** [comment]
And done.  Consider the long constant style slated for removal when v3.0 eventually rolls around.

**Pull request merged**